File: Number-of-Common-Factors.js
/**
 * @param {number} a
 * @param {number} b
 * @return {number}
 */
var commonFactors = function(a, b) {
    let minVal = Math.min(a, b);
    let count = 0;
    for (let i = 1; i <= minVal; i++) {
        if (a % i === 0 && b % i === 0) {
            count++;
        }
    }
    return count;
};

File: Number-of-Common-Factors.py
class Solution(object):
    def commonFactors(self, a, b):
        """
        :type a: int
        :type b: int
        :rtype: int
        """
        min_val = min(a, b)
        count = 0
        for i in range(1, min_val + 1):
            if a % i == 0 and b % i == 0:
                count += 1
        return count